core/editor/src/tools/ellipse.rs

```diff
@@ -1,12 +1,54 @@
-use crate::events::Event;
-use crate::tools::Tool;
+use crate::events::MouseKeys;
+use crate::events::{Event, Response};
+use crate::tools::{Fsm, Tool};
+use crate::Document;
 use document_core::Operation;
 
 #[derive(Default)]
-pub struct Ellipse;
+pub struct Ellipse {
+	state: EllipseToolState,
+}
 
 impl Tool for Ellipse {
-	fn handle_input(&mut self, event: Event) -> Vec<Operation> {
-		todo!();
+	fn handle_input(&mut self, event: &Event, document: &Document) -> (Vec<Response>, Vec<Operation>) {
+		let mut responses = Vec::new();
```

Comment:
I don't love the idea of returning data to the caller of `.transition()` by passing in mutable `Vec`s. Feels way too C-like. It's probably better that we just return a 3-tuple from `.transition()` with the two `Vec`s (transferring ownership to the caller) plus the current return value. We can destructure them in the caller.